concepts/box/about.md

```diff
@@ -1,3 +1,153 @@
 # About
 
-TODO
+To understand [`Box`][box] it is helpful to understand what the [stack and heap][stack and heap] are.
+
+The stack is an area of memory that works like a last-in-first-out (LIFO) buffer. It is very fast to push data onto the stack and it is fast to pop it off.
+The data can include variables or function addresses, but the data must be of a known fixed size at compilation time.
+
+If the data is of an unknown size at compilation time or has a size that may change, then it must be allocated on the heap. The allocation process must find
+space in the heap big enough to store the data, and then the address of where that data starts in memory gets pushed onto the stack. Retrieving data from the
+heap is slower because we need to go through the address on the stack and follow that to the location of tha data on the heap.
+
+`Box` is essentially a smart pointer which allocates data on the heap. The address is on the stack, pointing to the data on the heap. There are several
+[reasons][reasons] to use `Box` to put data on the heap:
+
+- When you have a type whose size can’t be known at compile time and you want to use a value of that type in a context that requires an exact size
+
+- When you have a large amount of data and you want to transfer ownership but ensure the data won’t be copied when you do so
+
+- When you want to own a value and you care only that it’s a type that implements a particular trait rather than being of a specific type
+
+A practical usage of `Box` is when you have multiple structs which you want to handle the same via a trait which they all implement.
+
+Let's say we have two structs. Each struct is of something that can be annoying. We can have each struct implement an Annoyance trait like so
+
+```rust
+struct WerewolfOfLondon {}
+struct Rooster {}
+
+trait Annoyance {
+    fn annoy(&self) -> &'static str;
+}
+
+impl Annoyance for WerewolfOfLondon {
+    fn annoy(&self) -> &'static str {
+        "ahhh-OOOOOOO!"
+    }
+}
+
+impl Annoyance for Rooster {
+    fn annoy(&self) -> &'static str {
+        "errr-errr-errr-errr-ERRRRRRRRR!"
+    }
+}
+```
+
+Now let's say we want a function that will give us an Annoyance based on the time of day that's passed into it. We can first try this, which we discover won't compile
+
+```rust
+enum TimeOfDay {
+    Midnight,
+    Morning,
+}
+
+//return something that implements Annoyance
+fn annoy_me(time_of_day: TimeOfDay) -> impl Annoyance {
+    match time_of_day {
+        TimeOfDay::Midnight => WerewolfOfLondon {},
+        TimeOfDay::Morning => Rooster {},
```

Comment:
```suggestion
        TimeOfDay::Midnight => WerewolfOfLondon,
        TimeOfDay::Morning => Rooster,
```

Assuming you took the previous unit struct optimization, you get a minor usability benefit here.

Also, the same edit in subsequent examples.